pkg/virtctl/create/vm/vm.go

```diff
@@ -779,3 +885,250 @@ func withCloudInitData(flag string, c *createVM, vm *v1.VirtualMachine) error {
+	sourceStruct := dataVolumeSourceBlank{}
+	if err := params.Map(VolumeImportFlag, paramStr, &sourceStruct); err != nil {
+		return nil, err
+	}
+
+	source := cdiv1.DataVolumeSource{
+		Blank: &cdiv1.DataVolumeBlankImage{},
+	}
+
+	return &source, nil
+}
+
+func withVolumeSourceHttp(paramStr string, vm *v1.VirtualMachine) (*cdiv1.DataVolumeSource, error) {
+	sourceStruct := dataVolumeSourceHttp{}
+	if err := params.Map(VolumeImportFlag, paramStr, &sourceStruct); err != nil {
+		return nil, err
+	}
+
+	if sourceStruct.URL == "" {
+		return nil, params.FlagErr(VolumeImportFlag, "URL is required with http volume source")
+	}
+
+	source := cdiv1.DataVolumeSource{
+		HTTP: &cdiv1.DataVolumeSourceHTTP{
+			URL:                sourceStruct.URL,
+			SecretRef:          sourceStruct.SecretRef,
+			CertConfigMap:      sourceStruct.CertConfigMap,
+			ExtraHeaders:       sourceStruct.ExtraHeaders,
+			SecretExtraHeaders: sourceStruct.SecretExtraHeaders,
+		},
+	}
+
+	return &source, nil
+}
+
+func withVolumeSourceImageIO(paramStr string, vm *v1.VirtualMachine) (*cdiv1.DataVolumeSource, error) {
+	sourceStruct := dataVolumeSourceImageIO{}
+	if err := params.Map(VolumeImportFlag, paramStr, &sourceStruct); err != nil {
+		return nil, err
+	}
+
+	if sourceStruct.URL == "" || sourceStruct.DiskId == "" {
+		return nil, params.FlagErr(VolumeImportFlag, "URL and diskid are both required with imageIO volume source")
+	}
+
+	source := cdiv1.DataVolumeSource{
+		Imageio: &cdiv1.DataVolumeSourceImageIO{
+			URL:           sourceStruct.URL,
+			DiskID:        sourceStruct.DiskId,
+			SecretRef:     sourceStruct.SecretRef,
+			CertConfigMap: sourceStruct.CertConfigMap,
+		},
+	}
+
+	return &source, nil
+}
+
+func withVolumeSourcePVC(paramStr string, vm *v1.VirtualMachine) (*cdiv1.DataVolumeSource, error) {
+	sourceStruct := dataVolumeSourcePVC{}
+	if err := params.Map(VolumeImportFlag, paramStr, &sourceStruct); err != nil {
+		return nil, err
+	}
+
+	if sourceStruct.Name == "" || sourceStruct.Namespace == "" {
+		return nil, params.FlagErr(VolumeImportFlag, "name and namespace are both required with PVC volume source")
+	}
+
+	source := cdiv1.DataVolumeSource{
+		PVC: &cdiv1.DataVolumeSourcePVC{
+			Name:      sourceStruct.Name,
+			Namespace: sourceStruct.Namespace,
+		},
+	}
+
+	return &source, nil
+}
+
+func withVolumeSourceRegistry(paramStr string, vm *v1.VirtualMachine) (*cdiv1.DataVolumeSource, error) {
+	sourceStruct := dataVolumeSourceRegistry{}
+	if err := params.Map(VolumeImportFlag, paramStr, &sourceStruct); err != nil {
+		return nil, err
+	}
+
+	if sourceStruct.PullMethod != "" &&
+		(sourceStruct.PullMethod != string(cdiv1.RegistryPullPod) && sourceStruct.PullMethod != string(cdiv1.RegistryPullNode)) {
+		return nil, params.FlagErr(VolumeImportFlag, "pullmethod must be set to pod or node")
+	}
+
+	if sourceStruct.URL == "" && sourceStruct.ImageStream == "" {
+		return nil, params.FlagErr(VolumeImportFlag, "at least on of url, imagestream must be defined")
```

Comment:
Typo: on

Reply:
You could merge this check and the one below and let it return `exactly one of url or imagestream must be defined`